homeassistant/components/rainmachine/switch.py

```diff
@@ -166,18 +181,34 @@ async def async_setup_entry(
     ]
     zones_coordinator = hass.data[DOMAIN][DATA_COORDINATOR][entry.entry_id][DATA_ZONES]
 
-    entities: list[RainMachineProgram | RainMachineZone] = []
-
-    for uid, program in programs_coordinator.data.items():
-        entities.append(
-            RainMachineProgram(
-                programs_coordinator, controller, uid, program["name"], entry
-            )
-        )
-    for uid, zone in zones_coordinator.data.items():
-        entities.append(
-            RainMachineZone(zones_coordinator, controller, uid, zone["name"], entry)
+    entities: list[RainMachineProgram | RainMachineZone] = [
+        RainMachineProgram(
+            programs_coordinator,
+            controller,
+            entry,
```

Comment:
No need to pass the entry, the coordinator already holds that.

Reply:
Does that feel a little too implicit? Passing the entry here very clearly shows that it comes from setup; using the reference from the coordinator seems a bit magical.

Reply:
Why add things that are already there? :man_shrugging: 
It's anyway not related to the change here :+1: